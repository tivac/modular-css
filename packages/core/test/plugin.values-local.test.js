"use strict";

var plugin = require("../plugins/values-local.js"),
    
    processor = require("postcss")([ plugin ]);

describe("/plugins", function() {
    describe("/values-local.js", function() {
        it("should ignore invalid declarations in normal mode", function() {
            expect(() => processor.process("@value red:").css).not.toThrow();
            expect(() => processor.process("@value blue red").css).not.toThrow();
        });
        
        it("should complain about invalid declarations in strict mode", function() {
            expect(
                () => processor.process("@value red:", { strict : true }).css
            )
            .toThrow(/SyntaxError: /);
            
            expect(
                () => processor.process("@value blue red", { strict : true }).css
            )
            .toThrow(/SyntaxError: /);
        });

        it("should remove value declarations from output", function() {
            expect(
                processor.process("@value red: #F00").css
            )
            .toMatchSnapshot();

            expect(
                processor.process("@value red: #F00; @value blue: blue;").css
            )
            .toMatchSnapshot();

            expect(
                processor.process("@value red: #F00; .wooga { color: red; }").css
            )
            .toMatchSnapshot();
        });
        
        it("should emit a message with details about values", function() {
            expect(
                processor.process("@value red: #F00; @value blue: blue;").messages
            )
            .toMatchSnapshot();
        });

        it("should ignore non-local values", function() {
            expect(
                processor.process(`@value red from "./fooga.css";`).css
            )
            .toMatchSnapshot();

            expect(
                processor.process(`@value red from "./fooga.css"; .fooga { color: red; }`).css
            )
            .toMatchSnapshot();
        });
    });
});
