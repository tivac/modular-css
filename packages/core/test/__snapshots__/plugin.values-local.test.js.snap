// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/plugins /values-local.js should emit a message with details about values 1`] = `
Array [
  Object {
    "plugin": "modular-css-values-local",
    "type": "modular-css",
    "values": Object {
      "blue": Object {
        "source": Object {
          "end": Object {
            "column": 36,
            "line": 1,
          },
          "input": Input {
            "css": "@value red: #F00; @value blue: blue;",
            "id": "<input css 8>",
          },
          "start": Object {
            "column": 19,
            "line": 1,
          },
        },
        "value": "blue",
      },
      "red": Object {
        "source": Object {
          "end": Object {
            "column": 17,
            "line": 1,
          },
          "input": Input {
            "css": "@value red: #F00; @value blue: blue;",
            "id": "<input css 8>",
          },
          "start": Object {
            "column": 1,
            "line": 1,
          },
        },
        "value": "#F00",
      },
    },
  },
]
`;

exports[`/plugins /values-local.js should ignore non-local values 1`] = `"@value red from \\"./fooga.css\\";"`;

exports[`/plugins /values-local.js should ignore non-local values 2`] = `"@value red from \\"./fooga.css\\"; .fooga { color: red; }"`;

exports[`/plugins /values-local.js should remove value declarations from output 1`] = `""`;

exports[`/plugins /values-local.js should remove value declarations from output 2`] = `""`;

exports[`/plugins /values-local.js should remove value declarations from output 3`] = `".wooga { color: red; }"`;
